lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -44,43 +44,56 @@ local function playerIsInBuild( ply )
     return !getPlayerPvpMode( ply )
 end
 
-local function movementMultiplier(weaponCount) 
+local function movementMultiplier( weaponCount )
+    if weaponCount < 1 then return 1 end
     return math.Clamp(1 - (1.9^( weaponCount))/100, 0, 1)
 end
 
-local function setSpeed(ply, multiplier) 
-    -- set speed multiplier, 0 - 1
-    ply:SetRunSpeed(math.Clamp(baseRunSpeed*multiplier,minRunSpeed, baseRunSpeed))
-    ply:SetWalkSpeed(math.Clamp(baseWalkSpeed*multiplier,minWalkSpeed, baseWalkSpeed))
+local function setSpeedFromWeaponCount( ply, weaponCount )
+    local multiplier = movementMultiplier( weaponCount )
+
+    local runSpeed = baseRunSpeed * multiplier
+    local walkSpeed = baseWalkSpeed * multiplier
+    ply:SetRunSpeed( math.max( runSpeed, minRunSpeed ) )
+    ply:SetWalkSpeed( math.max( walkSpeed, minWalkSpeed ) )
 end
 
-local function adjustMovementSpeed(ply) 
-    if playerIsInBuild( ply ) then 
-        setSpeed(ply, 1) 
-        return
+local function getWeaponWeight( weapon )
+    -- the weight/significance of a weapon (1 for affecting weight 0 for not affecting weight)
+    if not nonEffectedWeapons[weapon:GetClass()] then
```

Comment:
Can this logic be simplified?

```lua
local function getWeaponWeight( weapon )
    return nonEffectedWeapons[weapon:GetClass()]
end
```

But honestly, I think this should be refactored a little bit. This function's job should just be to get the weight of a given weapon, not decide whether or not it's affected by weapon weight.

I think this check should go down in the for loop in `getWeaponCount`

This function becomes extremely simple and could be replaced with a simple table existence check, but leaving this function here gives us the ability to expand it later.

Anyway, check out my comment in the other function.

Reply:
the primary reason i made this function was to simplify onEquip and onDrop. that check would have to be added to onEquip and onDrop as well

i think it makes sense to put that check in there as the "weight" of unaffected weapons is 0

Reply:
```lua
local function getWeaponWeight( weapon )
    -- the weight/significance of a weapon (1 for affecting weight 0 for not affecting weight)
    if nonEffectedWeapons[weapon:GetClass()] then return 0 end

    return 1
end
```
what do you think about just refactoring it like this?